build/lib/locFunc.js

```diff
@@ -41,3 +49,282 @@ function packageSingleExtensionStream(name) {
+        'Do not edit this file. It is machine generated.'
+    ];
+    for (let key of Object.keys(messages)) {
+        result[key] = messages[key];
+    }
+    let content = JSON.stringify(result, null, '\t');
+    if (process.platform === 'win32') {
+        content = content.replace(/\n/g, '\r\n');
+    }
+    return new File({
+        path: path.join(originalFilePath + '.i18n.json'),
+        contents: Buffer.from(content, 'utf8'),
+    });
+}
+/**
+ * Function handles the processing of xlf resources and turning them into i18n.json files.
+ * It adds the i18n files translation paths to be added back into package.main.
+ * Based on prepareI18nPackFiles in i18n.ts
+*/
+function modifyI18nPackFiles(existingTranslationFolder, resultingTranslationPaths, pseudo = false) {
+    let parsePromises = [];
+    let mainPack = { version: i18n.i18nPackVersion, contents: {} };
+    let extensionsPacks = {};
+    let errors = [];
+    return (0, event_stream_1.through)(function (xlf) {
+        let rawResource = path.basename(xlf.relative, '.xlf');
+        let resource = rawResource.substring(0, rawResource.lastIndexOf('.'));
+        let contents = xlf.contents.toString();
+        let parsePromise = pseudo ? i18n.XLF.parsePseudo(contents) : i18n.XLF.parse(contents);
+        parsePromises.push(parsePromise);
+        parsePromise.then(resolvedFiles => {
+            resolvedFiles.forEach(file => {
+                const path = file.originalFilePath;
+                const firstSlash = path.indexOf('/');
+                //exclude core sql file from extension processing.
+                if (resource !== 'sql') {
+                    let extPack = extensionsPacks[resource];
+                    if (!extPack) {
+                        extPack = extensionsPacks[resource] = { version: i18n.i18nPackVersion, contents: {} };
+                    }
+                    //remove extensions/extensionId section as all extensions will be webpacked.
+                    const secondSlash = path.indexOf('/', firstSlash + 1);
+                    extPack.contents[path.substr(secondSlash + 1)] = file.messages;
+                }
+                else {
+                    mainPack.contents[path.substr(firstSlash + 1)] = file.messages;
+                }
+            });
+        }).catch(reason => {
+            errors.push(reason);
+        });
+    }, function () {
+        Promise.all(parsePromises)
+            .then(() => {
+            if (errors.length > 0) {
+                throw errors;
+            }
+            const translatedMainFile = updateMainI18nFile(existingTranslationFolder + '\\main', './main', mainPack);
+            this.queue(translatedMainFile);
+            for (let extension in extensionsPacks) {
+                const translatedExtFile = i18n.createI18nFile(`extensions/${extension}`, extensionsPacks[extension]);
+                this.queue(translatedExtFile);
+                //handle edge case for 'Microsoft.sqlservernotebook'
```

Comment:
What "edge case"?

Reply:
The edge case is that the file name for the Microsoft.sqlservernotebook extension is literally "Microsoft.sqlservernotebook" and that it also has that id instead of "Microsoft.Microsoft.sqlservernotebook". This line is to handle this issue.

Reply:
Make sure this is documented in the comment

Reply:
Done so.